server/controller/src/main/java/ai/starwhale/mlops/api/SWModelPackageController.java

```diff
@@ -20,49 +36,122 @@
-        return null;
+    public ResponseEntity<ResponseMessage<PageInfo<SWModelPackageVersionVO>>> listModelVersion(String projectId,
+        String modelId, String modelVersionName, Integer pageNum, Integer pageSize) {
+        List<SWModelPackageVersionVO> voList = swmpService.listSWMPVersionHistory(
+            SWMPObject.builder()
+                .projectId(projectId)
+                .id(modelId)
+                .latestVersion(Version.builder().name(modelVersionName).build())
+                .build(),
+            PageParams.builder()
+                .pageNum(pageNum)
+                .pageSize(pageSize)
+                .build());
+        PageInfo<SWModelPackageVersionVO> pageInfo = new PageInfo<>(voList);
+        return ResponseEntity.ok(Code.success.asResponse(pageInfo));
     }
 
     @Override
-    public ResponseEntity<ResponseMessage<String>> createModelVersion(String modelId,
+    public ResponseEntity<ResponseMessage<String>> createModelVersion(String projectId, String modelId,
         MultipartFile zipFile, String importPath) {
-        return null;
+        UserVO user = userService.currentUser();
+        String versionId = createVersion(projectId, modelId, zipFile, importPath, user.getId());
+        return ResponseEntity.ok(Code.success
+            .asResponse(String.valueOf(Optional.of(versionId).orElseThrow(ApiOperationException::new))));
     }
 
     @Override
-    public ResponseEntity<ResponseMessage<String>> modifyModel(String modelId, String versionId,
+    public ResponseEntity<ResponseMessage<String>> modifyModel(String projectId, String modelId, String versionId,
         String tag) {
-        return null;
+        Boolean res = swmpService.modifySWMPVersion(
+            Version.builder().id(versionId).tag(tag).build());
+        Assert.isTrue(Optional.of(res).orElseThrow(ApiOperationException::new));
+        return ResponseEntity.ok(Code.success.asResponse("success"));
     }
 
     @Override
-    public ResponseEntity<ResponseMessage<String>> createModel(String modelName,
+    public ResponseEntity<ResponseMessage<String>> createModel(String projectId, String modelName,
         MultipartFile zipFile, @RequestParam(value = "importPath") String importPath) {
-        return null;
+        UserVO user = userService.currentUser();
+        String modelId = swmpService.addSWMP(
+            SWMPObject.builder().projectId(projectId).name(modelName).ownerId(user.getId())
+                .build());
+        String versionId = createVersion(projectId, modelId, zipFile, importPath, user.getId());
+        return ResponseEntity.ok(Code.success.asResponse(String.valueOf(Optional.of(versionId).orElseThrow(ApiOperationException::new))));
+    }
+
+    private String createVersion(String projectId, String modelId, MultipartFile zipFile, String importPath, String userId) {
+        String path = importPath;
+        String meta = "";
+        if (zipFile != null) {
```

Comment:
swmp now use tar format, no compress

Reply:
I see. This part of the business has not been completed yet. This is just to get through the front-end interaction process